src/webgpu/util/texture/texel_data.spec.ts

```diff
@@ -106,16 +107,67 @@ function doTest(
   pass.end();
   t.device.queue.submit([encoder.finish()]);
 
-  t.expectGPUBufferValuesEqual(
-    outputBuffer,
-    new ReadbackTypedArray(
-      rep.componentOrder.map(c => {
-        const value = rep.decode(componentData)[c];
-        assert(value !== undefined);
-        return value;
-      })
-    )
+  const idealReadbackData = new ReadbackTypedArray(
+    rep.componentOrder.map(c => {
+      const value = rep.decode(componentData)[c];
+      assert(value !== undefined);
+      return value;
+    })
   );
+
+  if (format === 'rgba8unorm-srgb' || format === 'bgra8unorm-srgb') {
+    // In D3D the SRGB -> Float conversion is permitted a tolerance of 0.5f ULP on the SRGB side.
+    // The procedure for measuring this tolerance is to convert the result back into SRGB space
+    // using the ideal FLOAT -> SRGB conversion specified below but WITHOUT the rounding to integer,
+    // and taking the floating point difference versus the original SRGB value to yield the error.
+    // Exact conversion is required: 0.0f and 1.0f (the ends) must be exactly achievable.
+    // https://microsoft.github.io/DirectX-Specs/d3d/archive/D3D11_3_FunctionalSpec.htm#3.2.3.7%20SRGB%20-%3E%20FLOAT
+    const readBackValue = await t.readGPUBufferRangeTyped(outputBuffer, {
+      type: Float32Array,
+      typedLength: 4,
+    });
+    t.expect(idealReadbackData[3] === readBackValue.data[3]);
+
+    // Do the ideal FLOAT -> SRGB conversion on the readback float value WITHOUT the rounding to
+    // integer.
+    const readBackValueToSRGB = new Float32Array(3);
+    for (let i = 0; i < 3; ++i) {
+      const gamaCompressed = gammaCompress(readBackValue.data[i]);
+      let outputIndex = i;
+      if (format === 'bgra8unorm-srgb' && (i === 0 || i === 2)) {
+        outputIndex = 2 - i;
+      }
+      readBackValueToSRGB[outputIndex] = floatAsNormalizedIntegerScale(gamaCompressed, 8, false);
+    }
+
+    // Take the floating point difference versus the original SRGB value to yield the error.
+    const toleranceULP = 0.5;
```

Comment:
Should this be _slightly_ larger than 0.5 to allow for error on the JS side? (Though, if it passes, maybe it doesn't matter)

Reply:
I just keep using `0.5` as all the tests can pass right now. What do you think?